lib/Doctrine/DBAL/Query/Expression/CompositeExpression.php

```diff
@@ -43,47 +47,48 @@ class CompositeExpression implements Countable
      */
     public function __construct(string $type, array $parts = [])
     {
-        $this->type = $type;
-
-        $this->addMultiple($parts);
+        $this->type  = $type;
+        $this->parts = array_values(array_filter($parts, static function ($part) {
+            return ! ($part instanceof self && count($part) === 0);
+        }));
     }
 
     /**
-     * Adds multiple parts to composite expression.
+     * Returns a CompositeExpression with the given parts added.
      *
-     * @param array<int, self|string> $parts
+     * This instance is immutable and unaffected by this method call.
      *
-     * @return $this
+     * @param array<int, self|string> $parts
      */
     public function addMultiple(array $parts = []) : self
```

Comment:
Let's remove and make `with()` variadic instead. There's no longer point in having two methods for the same operation.

Reply:
Done!